packages/dapp/src/components/KeyFormModal.tsx

```diff
@@ -0,0 +1,140 @@
+import type { AnySchema } from '@windingtree/org.id-utils/dist/object';
+import { useState, FC, useEffect } from 'react';
+import { Box, Button, Form, FormField, TextInput } from 'grommet';
+import { useKeysManager, allowedKeysTypes, revocationReasonValues } from '../hooks/useKeysManager';
+import { useAppState } from '../store';
+import { KeyRecordRaw } from '../store/actions';
+import { Modal } from './Modal';
+import { object } from '@windingtree/org.id-utils';
+
+export const keyRecordRawSchema: AnySchema = {
+  type: 'object',
+  properties: {
+    type: {
+      type: 'string',
+      enum: allowedKeysTypes
+    },
+    publicKey: {
+      type: 'string'
+    },
+    tag: {
+      type: 'string'
+    },
+    note: {
+      type: 'string'
+    },
+    revocation: {
+      type: 'object',
+      properties:{
+        reason: {
+          type: 'string',
+          enum: revocationReasonValues
+        },
+        invalidityDate: {
+          type: 'string',
+          format: 'date-time'
+        },
+      },
+      required: ['reason', 'invalidityDate']
+    }
+  },
+  required: ['type', 'publicKey', 'tag']
+}
+
+const validateKeyRecordRawWithSchema = (record: KeyRecordRaw): string | null =>
+  object.validateWithSchemaOrRef(
+    keyRecordRawSchema,
+    '',
+    record
+  );
+
+const newRecord:KeyRecordRaw = {
+  tag: '',
+  publicKey:'',
+  note:'',
+  type: allowedKeysTypes[0]
+}
+
+export const KeyFormModal:FC<{close():void}> = ({close, children}) => {
+  const { account, keys, isConnecting } = useAppState();
+  const [addKey] = useKeysManager()
+  const [rawRecord, setRawRecord] = useState(newRecord);
+  const [error, setError] = useState<string | undefined>();
```

Comment:
This error is not displayed anywhere but should.
In my task, I have created a unified component for displaying error messages.
For now, just add `<Text>` with an error message under the form. When my PR will be merged you will be able to update the code to use my Message component.

Reply:
✔️